### PR TITLE
Add blog post about cache support collab

### PR DESCRIPTION
**Pull request by mosabua**
Will change date when its ready to go...

### CODE REVIEW

File: _posts/2024-02-26-cache-refresh.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+title:  "A cache refresh for Trino"
+author: Manfred Moser
+image: /assets/blog/trino-cache-refresh.png
+excerpt_separator: <!--more-->
+show_pagenav: true
+---
+
+Leon Bambrick once said "There are only two hard things in computer science:
+cache invalidation, naming things, and off-by-one errors" improving the initial
+wisdom from [Phil Karlton](https://www.karlton.org/2017/12/naming-things-hard/).
```
Comment (colebow) [inline _posts/2024-02-26-cache-refresh.md:12]: I'm a little uncertain about this quote and citing two different individuals, as well as linking to this kinda sketchy website. I'm not familiar with the person we're referencing, and I don't think there's an expectation for them to be well-known names, so proper writing would need to introduce or explain background on them.

But something like, "There's a famous proverb: 'There are only two hard things...'" without attribution is actually fair game, as the quote really is a proverb at this point and [thus doesn't need to be attributed](https://style.mla.org/citing-famous-sayings/).

The Leon Bambrick variation isn't as much a proverb, and I'd think we can just drop the "off by one errors" portion of the quote to avoid the messiness of attributing it to a not-super-notable individual and needing to explain who they are.

File: _posts/2024-02-26-cache-refresh.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+Well, in the Trino community we know all about caching, naming, off-by-one
+errors, and improving on original wisdom. With the recent [Trino 439
+release]({{site.url}}/docs/current/release/release-438.html) caching from object
```
Comment (colebow) [inline _posts/2024-02-26-cache-refresh.md:15]:
```suggestion
release]({{site.url}}/docs/current/release/release-438.html), caching from object
```

File: _posts/2024-02-26-cache-refresh.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+storage file systems got a refresh. Users of catalogs using the Delta Lake,
```
Comment (colebow) [inline _posts/2024-02-26-cache-refresh.md:16]: "Users of catalogs using" sounds kind of redundant, I'd reword.

File: _posts/2024-02-26-cache-refresh.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+Hive, and Iceberg connectors now get to enjoy performance benefits from the new
+Alluxio-powered file system caching. And all that as a replacement from the now
+removed object storage caching with RubiX.
```
Comment (colebow) [inline _posts/2024-02-26-cache-refresh.md:19]: This last sentence is a fragment - maybe say, "In case you're wondering about the old caching we had in Hive, this is replacing..."

File: _posts/2024-02-26-cache-refresh.md [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+<!--more-->
+
+## In the past
+
+So how did we get here? A long, long time ago Quoble open sourced a [light
```
Comment (colebow) [inline _posts/2024-02-26-cache-refresh.md:25]: A 3-for-1 grammar nitpick may be my new record
```suggestion
So how did we get here? A long, long time ago, Qubole open-sourced a [light
```

File: _posts/2024-02-26-cache-refresh.md [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+light-weight data caching framework called
+RubiX](https://github.com/qubole/rubix). The library was integrated into the
+Trino Hive connector and enabled [Hive connector storage
```
Comment (colebow) [inline _posts/2024-02-26-cache-refresh.md:28]:
```suggestion
Trino Hive connector, and it enabled [Hive connector storage
```

File: _posts/2024-02-26-cache-refresh.md [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+caching]({{site.url}}/docs/438/connector/hive-caching.html). However, over time
```
Comment (colebow) [inline _posts/2024-02-26-cache-refresh.md:29]: We need a comma after "over time," so let's change "However" to "But" to simplify the flow and not have too many commas breaking things up.
```suggestion
caching]({{site.url}}/docs/438/connector/hive-caching.html). But over time,
```

File: _posts/2024-02-26-cache-refresh.md [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+any open source project without active maintenance becomes stale. And like a
+stale cache, a stale open source project can cause issues, or at a minimum
```
Comment (colebow) [inline _posts/2024-02-26-cache-refresh.md:31]:
```suggestion
stale cache, a stale open source project can cause issues, or
```

File: _posts/2024-02-26-cache-refresh.md [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+becomes outdated and unsuitable for modern use. Staleness and other problems
+meant that RubiX had to go.
```
Comment (colebow) [inline _posts/2024-02-26-cache-refresh.md:33]: I think we could phrase this a little lighter, or even insert some humor. Whether that's, "though RubiX had once served Trino well, it was its time for it to go," or referencing it going to the great software repo in the sky, I leave to you.

File: _posts/2024-02-26-cache-refresh.md [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+
+## Making progress
+
+Time moved on and Trino now includes powerful connectors for the modern
```
Comment (colebow) [inline _posts/2024-02-26-cache-refresh.md:37]:
```suggestion
Catching back up to 2024, Trino now includes powerful connectors for the modern
```
Or something similar, "time moved on" sounds a little melodramatic.

File: _posts/2024-02-26-cache-refresh.md [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+</div>
+
+Hive is still around, just like HDFS, but we consider them both close to legacy
+status. Yet all four connectors could benefit from caching. Good news came at
+Trino Summit 2022 when Hope Wang and Beinan Wang from
+[Alluxio]({{site.url}}/ecosystem/add-on.html#alluxio) presented about their
+integration with Trino and the Hive connector - [Trino optimization with
+distributed caching on data lake]({% post_url
+2023-07-21-trino-fest-2023-alluxio-recap %}). They mentioned plans to open
+source their implementation and an initial PR was created.
```
Comment (colebow) [inline _posts/2024-02-26-cache-refresh.md:67]: Move `pull request (PR)` up here, as this is the first use of the term.

File: _posts/2024-02-26-cache-refresh.md [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+    <div class="col-sm">
+      <img src="{{site.url}}/assets/images/logos/alluxio.png" title="Alluxio">
+    </div>
+    <div class="col-sm"></div>
+  </div>
+</div>
+
+## Collaboration
+
+The initial presentation and pull request (PR) planted a seed in the community.
```
Comment (colebow) [inline _posts/2024-02-26-cache-refresh.md:81]: And just go with `PR` down here.

File: _posts/2024-02-26-cache-refresh.md [hunk 13]
```diff
@@ -0,0 +1,166 @@
...
+The Trino project had been moving fast in terms of deprecating the old
+dependencies from the Hadoop and Hive ecosystem so the Alluxio PR was not ready
```
Comment (colebow) [inline _posts/2024-02-26-cache-refresh.md:83]:
```suggestion
dependencies from the Hadoop and Hive ecosystem, so the Alluxio PR was not ready
```
Is there a more specific (but still diplomatic) way to say "not ready for a merge?"

File: _posts/2024-02-26-cache-refresh.md [hunk 14]
```diff
@@ -0,0 +1,166 @@
...
+for a merge. Discussions with [David Phillips](https://github.com/electrum) laid
+out the path to adjust to the new file system support.
+
+In the end it was [Florent Delannoy](https://github.com/pluies) who started
+another [PR for file system caching support, specifically for the Delta Lake
+connector](https://github.com/trinodb/trino/pull/18719). His team mate [Jonas
```
Comment (colebow) [inline _posts/2024-02-26-cache-refresh.md:89]:
```suggestion
connector](https://github.com/trinodb/trino/pull/18719). His teammate [Jonas
```

File: _posts/2024-02-26-cache-refresh.md [hunk 15]
```diff
@@ -0,0 +1,166 @@
...
+Irgens Kylling](https://github.com/jkylling), also a [presenter from Trino Fest
+  2023]({% post_url 2023-07-14-trino-fest-2023-dune %}), took over the work on
+the PR. The collaboration on it was an **epic effort**. Over 300
+comments directly on GitHub and numerous hours of coding, reviewing, testing,
+and discussion on slack and elsewhere finally ended with a merge after many
```
Comment (colebow) [inline _posts/2024-02-26-cache-refresh.md:94]:
```suggestion
and discussion on Slack and elsewhere finally ended with a merge after many
```
"ended with a merge" feels anticlimactic, maybe specify that this means the launch of the feature and it shipping live with Trino

File: _posts/2024-02-26-cache-refresh.md [hunk 16]
```diff
@@ -0,0 +1,166 @@
...
+Phillips](https://github.com/electrum), [Raunaq
+Morarka](https://github.com/raunaqmorarka), [Piotr
+Findeisen](https://github.com/findepi), [Mateusz
+Gajewski](https://github.com/wendigo), [Beinan Wang](https://github.com/beinan),
+[Amogh Margoor](https://github.com/amoghmargoor), and [Marton
+Bod](https://github.com/marton-bod).
+
+## Finishing
+
+In parallel to the work on the initial PR for Delta Lake, I ended working on the
```
Comment (colebow) [inline _posts/2024-02-26-cache-refresh.md:107]:
```suggestion
In parallel to the work on the initial PR for Delta Lake, I ended up working on the
```

Comment (colebow) [reply _posts/2024-02-26-cache-refresh.md:107]: Maybe say "yours truly" instead of "I," which is a little jarring for a blog that isn't otherwise written in first person.

File: _posts/2024-02-26-cache-refresh.md [hunk 17]
```diff
@@ -0,0 +1,166 @@
...
+Margoor](https://github.com/amoghmargoor) sent a PR for Iceberg support. They
+were also about to add Hive support, when [Raunaq
+Morarka](https://github.com/raunaqmorarka) beat them and submitted that PR.
+
+After some final clean up, [Cole Bowden](https://github.com/colebow) and [Martin
+Traverso](https://github.com/martint) got the release notes together and shipped
+[Trino 439]({{site.url}}/docs/current/release/release-438.html)! Now you can use
+it too.
+
+## Using file system cache
```
Comment (colebow) [inline _posts/2024-02-26-cache-refresh.md:126]: "the file system cache" or "file system caching"?

File: _posts/2024-02-26-cache-refresh.md [hunk 18]
```diff
@@ -0,0 +1,166 @@
...
+properties]({{site.url}}/docs/current/object-storage/file-system-cache.html).
+
+## What's next
+
+What a success! It took many members from the global Trino village to get this
+feature added. Now our users across the globe can enjoy even more benefits of
+using Trino, and also participate in our next steps:
+
+* Further improvements to the current implementation, maybe adding worker to
+  worker connections for exchanging cached files.
```
Comment (colebow) [inline _posts/2024-02-26-cache-refresh.md:152]: worker-to-worker

**Review by colebow — CHANGES_REQUESTED** [review]

File: _posts/2024-02-26-cache-refresh.md (hunk 1 above)
Comment (mosabua) [reply _posts/2024-02-26-cache-refresh.md:12]: The website is not sketchy at all.. just old. And the two authors are both famous and I think we want to at least attribute to the Karlton site.. and it has lots more links to related stuff. I chose the Bambrick quote because its funny..  so it makes for a light entry to the blog post

File: _posts/2024-02-26-cache-refresh.md (hunk 3 above)
Comment (mosabua) [reply _posts/2024-02-26-cache-refresh.md:16]: done

File: _posts/2024-02-26-cache-refresh.md (hunk 4 above)
Comment (mosabua) [reply _posts/2024-02-26-cache-refresh.md:19]: I am removing the sentence.

File: _posts/2024-02-26-cache-refresh.md (hunk 7 above)
Comment (mosabua) [reply _posts/2024-02-26-cache-refresh.md:29]: Done

File: _posts/2024-02-26-cache-refresh.md (hunk 9 above)
Comment (mosabua) [reply _posts/2024-02-26-cache-refresh.md:33]: Updated

File: _posts/2024-02-26-cache-refresh.md (hunk 11 above)
Comment (mosabua) [reply _posts/2024-02-26-cache-refresh.md:67]: Done

File: _posts/2024-02-26-cache-refresh.md (hunk 13 above)
Comment (mosabua) [reply _posts/2024-02-26-cache-refresh.md:83]: Reworded.

File: _posts/2024-02-26-cache-refresh.md (hunk 15 above)
Comment (mosabua) [reply _posts/2024-02-26-cache-refresh.md:94]: Reworded

File: _posts/2024-02-26-cache-refresh.md (hunk 16 above)
Comment (mosabua) [reply _posts/2024-02-26-cache-refresh.md:107]: sounds good

**Comment by mosabua** [comment]
Adjusted to all feedback from @colebow now.

File: _posts/2024-02-26-cache-refresh.md [hunk 19]
```diff
@@ -0,0 +1,165 @@
+---
+layout: post
+title:  "A cache refresh for Trino"
+author: Manfred Moser
+image: /assets/blog/trino-cache-refresh.png
+excerpt_separator: <!--more-->
+show_pagenav: true
+---
+
+Thinking about our recent work on caching in Trino reminds me of he famous
```
Comment (colebow) [inline _posts/2024-02-26-cache-refresh.md:10]:
```suggestion
Thinking about our recent work on caching in Trino reminds me of the famous
```

File: _posts/2024-02-26-cache-refresh.md [hunk 20]
```diff
@@ -0,0 +1,165 @@
...
+saying [There are only two hard things in computer science: cache invalidation
+and naming things](https://www.karlton.org/2017/12/naming-things-hard/). Well,
```
Comment (colebow) [inline _posts/2024-02-26-cache-refresh.md:12]:
```suggestion
saying, ["There are only two hard things in computer science: cache invalidation
and naming things](https://www.karlton.org/2017/12/naming-things-hard/)." Well,
```

**Review by colebow — APPROVED** [review]
I think this is good to go with these edits

File: _posts/2024-02-26-cache-refresh.md [hunk 21]
```diff
@@ -0,0 +1,165 @@
...
+[Trino 439 release]({{site.url}}/docs/current/release/release-439.html), caching
+from object storage file systems got a refresh. Catalogs using the Delta Lake,
+Hive, and Iceberg connectors now get to access performance benefits from the new
+Alluxio-powered file system caching.
+
+<!--more-->
+
+## In the past
+
+So how did we get here? A long, long time ago, Quoble open-sourced a [light
```
Comment (sajjoseph) [inline _posts/2024-02-26-cache-refresh.md:23]: typo: Qubole

Comment (mosabua) [reply _posts/2024-02-26-cache-refresh.md:23]: Thanks .. fixed

File: _posts/2024-02-26-cache-refresh.md [hunk 22]
```diff
@@ -0,0 +1,165 @@
...
+* Enable file system caching and configure the cache location, for example at
+  `/tmp/trino-cache` on the nodes for in your catalog properties files.
+
+```
+fs.cache.enabled=true
+fs.cache.directories=/tmp/trino-cache
+```
+
+After a cluster restart, file system caching is active for the configured
+catalogs. You can  and you can tweak it with [further, optional configuration
```
Comment (dain) [inline _posts/2024-02-26-cache-refresh.md:141]: double space here?

**Review by dain — APPROVED** [review]

File: _posts/2024-02-26-cache-refresh.md (hunk 22 above)
Comment (mosabua) [reply _posts/2024-02-26-cache-refresh.md:141]: 🦅 👁️ @dain

File: _posts/2024-03-07-cache-refresh.md [hunk 23]
```diff
@@ -0,0 +1,165 @@
...
+
+So how did we get here? A long, long time ago, Qubole open-sourced a [light
+light-weight data caching framework called
+RubiX](https://github.com/qubole/rubix). The library was integrated into the
+Trino Hive connector, and it enabled [Hive connector storage
+caching]({{site.url}}/docs/438/connector/hive-caching.html). But over time, any
+open source project without active maintenance becomes stale. And like a stale
+cache, a stale open source project can cause issues, or becomes outdated and
+unsuitable for modern use. Though RubiX had once served Trino well, it was time
+to remove the dust, RubiX had to go.
```
Comment (electrum) [inline _posts/2024-03-07-cache-refresh.md:31]: Grammar is off. Last part seems to be missing "and"

File: _posts/2024-03-07-cache-refresh.md [hunk 24]
```diff
@@ -0,0 +1,165 @@
...
+show_pagenav: true
+---
+
+Thinking about our recent work on caching in Trino reminds me of the famous
+saying, ["There are only two hard things in computer science: cache invalidation
+and naming things](https://www.karlton.org/2017/12/naming-things-hard/)." Well,
+in the Trino community we know all about caching and naming. With the recent
+[Trino 439 release]({{site.url}}/docs/current/release/release-439.html), caching
+from object storage file systems got a refresh. Catalogs using the Delta Lake,
+Hive, and Iceberg connectors now get to access performance benefits from the new
```
Comment (electrum) [inline _posts/2024-03-07-cache-refresh.md:16]: We mention Hudi below. Should we add "(and soon Hudi)" or something?

File: _posts/2024-03-07-cache-refresh.md [hunk 25]
```diff
@@ -0,0 +1,165 @@
...
+
+[Manish Malhorta](https://github.com/osscm) and [Amogh
+Margoor](https://github.com/amoghmargoor) sent a PR for Iceberg support. They
+were also about to add Hive support, when [Raunaq
+Morarka](https://github.com/raunaqmorarka) beat them and submitted that PR.
+
+After some final clean up, [Cole Bowden](https://github.com/colebow) and [Martin
+Traverso](https://github.com/martint) got the release notes together and shipped
+[Trino 439]({{site.url}}/docs/current/release/release-438.html)! Now you can use
+it too.
```
Comment (electrum) [inline _posts/2024-03-07-cache-refresh.md:123]: Comma before "too"?

File: _posts/2024-03-07-cache-refresh.md [hunk 26]
```diff
@@ -0,0 +1,165 @@
...
+
+## Using file system caching
+
+There are only a few relatively simple steps to add file system caching to your
+catalogs that use Delta Lake, Hive, or Iceberg connectors:
+
+* Provision fast local file system storage on all your Trino cluster nodes. How
+  you do that depends on your cluster provisioning.
+* Enable file system caching and configure the cache location, for example at
+  `/tmp/trino-cache` on the nodes for in your catalog properties files.
```
Comment (electrum) [inline _posts/2024-03-07-cache-refresh.md:133]: Extra word "for"

**Review by electrum — APPROVED** [review]

File: _posts/2024-03-07-cache-refresh.md (hunk 24 above)
Comment (mosabua) [reply _posts/2024-03-07-cache-refresh.md:16]: yes .. done

File: _posts/2024-03-07-cache-refresh.md (hunk 23 above)
Comment (mosabua) [reply _posts/2024-03-07-cache-refresh.md:31]: yeah .. could have been "and" .. or long dash ... opted for and

File: _posts/2024-03-07-cache-refresh.md (hunk 26 above)
Comment (mosabua) [reply _posts/2024-03-07-cache-refresh.md:133]: fixed ..

**Comment by mosabua** [comment]
Also chatted with @martint in meeting and he had no concerns with merge.
